### PR TITLE
Allowing modification of throughput in Spark

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
A pretty common ask from customers is to allow changing a container's (or database's) throughput in Spark directly via `ALTER TABLE` or `ALTER DATABASE` command - without having to use a language-specific (PySpark -> Cosmos Python SDK, Scala -> Cosmos Java SDK, .Net -> Cosmos .Net SDK) additional dependency to do so. This ask gets even more pronounced when it means not having to take a dependency to one - but two - SDKs when trying to support changing the throughput in both key and AAD based authentication (the latter needs management SDK instead of "normal" data plane SDK).
And given that scaling-up at the beginning of a notebook or a job and  down at the end is a pretty often used pattern in Spark this PR is adding support to change the throughput via `ALTER TABLE` or `ALTER DATABASE` commands.

## Sample 01: Container-level setting throughput with manual throughput to 8,000 RU/s
```sql
ALTER TABLE cosmosCatalog.myDatabase.myContainer SET TBLPROPERTIES('manualThroughput' = '8000')
```

## Sample 02: Container-level setting throughput with max. auto-scale throughput to 4,000 RU/s
```sql
ALTER TABLE cosmosCatalog.myDatabase.myContainer SET TBLPROPERTIES('autoScaleMaxThroughput'= '4000')
```

## Sample 03: Shared-DB-level setting throughput with manual throughput to 8,000 RU/s
```sql
ALTER DATABASE cosmosCatalog.myDatabase SET DBPROPERTIES('manualThroughput' = '8000')
```

## Sample 04: Shared-DB-level setting throughput with max. auto-scale throughput to 4,000 RU/s
```sql
ALTER DATABASE cosmosCatalog.myDatabase  SET DBPROPERTIES('autoScaleMaxThroughput'= '4000')
```

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 #### Features Added
 * Added Service Principle based AAD Auth - See [PR 32393](https://github.com/Azure/azure-sdk-for-java/pull/32393)
+* Added capability to allow modification of throughput in Spark via `ALTER TABLE` or `ALTER DATABASE` command. - See [PR 32393](https://github.com/Azure/azure-sdk-for-java/pull/33369)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:7]: the PR number reference is not correct

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala [hunk 2]
```diff
@@ -412,4 +415,102 @@ private[spark] case class CosmosCatalogManagementSDKClient(resourceGroupName: St
...
+
+        sqlResourcesClient
+          .updateSqlDatabaseThroughputAsync(
+            resourceGroupName,
+            databaseAccountName,
+            databaseName,
+            throughputUpdateParameter)
+          .asScala
+          .map(result => {
+            if (!result.resource().offerReplacePending().toBoolean) {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala:504]: Do we need to also check the following?
```
 if (result.resource().offerReplacePending() != null &&
                    result.resource().offerReplacePending() != "null" &&
```

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogManagementSDKClient.scala]: Done

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md]: Fixed

**Pull request merged**